### PR TITLE
initial working solution to slow members refresh service

### PR DESCRIPTION
**Pull request by abarker66**
Added several improvements to the Members' Refresh service.

- Handles missing endpoints correctly
- Async calls to the endpoints
- Bulk repo updates rather than single operations.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/info/client/ingest/ListConsumerService.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public ListConsumerService(@Value("${api.uk.parliament}") String apiUkParliament
                                     houseAddress.getUuid(),
                                     "NI" + m.getPersonId()))
                     .collect(Collectors.toSet());
-        } catch (ApplicationExceptions.EntityNotFoundException ex) {
+        } catch (ApplicationExceptions.IngestException ex) {
             log.info(ex.getMessage());
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/client/ingest/ListConsumerService.java:76]: Would `log.warn` be more suitable?

File: src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java [hunk 2]
```diff
@@ -42,28 +46,50 @@ Member getMember(UUID uuid) {
...
         log.info("Started Updating Members Lists");
-        updateMember(listConsumerService.createFromWelshAssemblyAPI());
-        updateMember(listConsumerService.createFromScottishParliamentAPI());
-        updateMember(listConsumerService.createCommonsFromUKParliamentAPI());
-        updateMember(listConsumerService.createLordsFromUKParliamentAPI());
-        updateMember(listConsumerService.createFromIrishAssemblyAPI());
+        Long start = System.currentTimeMillis();
+
+        // launch async calls to all members endpoints, threads taken from common pool.
+        CompletableFuture<Set<Member>> futureWelsh
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:54]: Really like this!

File: src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java [hunk 3]
```diff
@@ -42,28 +46,50 @@ Member getMember(UUID uuid) {
...
 
-    void updateWebMemberLists() {
+    public Integer updateWebMemberLists() {
         log.info("Started Updating Members Lists");
-        updateMember(listConsumerService.createFromWelshAssemblyAPI());
-        updateMember(listConsumerService.createFromScottishParliamentAPI());
-        updateMember(listConsumerService.createCommonsFromUKParliamentAPI());
-        updateMember(listConsumerService.createLordsFromUKParliamentAPI());
-        updateMember(listConsumerService.createFromIrishAssemblyAPI());
+        Long start = System.currentTimeMillis();
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:51]: Is this valid for when in an environment or was it just for local testing?

Comment (abarker66) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:51]: Yes that can go, wanted to collect metrics during dev and would like to know how it behaves in prod but it is not essential.

Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:51]: Could use the log above timestamp with the finishing one for a rough guess. Won't be fully accurate though...

**Review by LiamMacP — APPROVED** [review]
Approved - looks good and a much more flexible solution.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java (hunk 2 above)
Comment (abarker66) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:54]: Ha ha, I see, yes - I'm with you - thought you were having a pop at my lovely code.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java (hunk 3 above)
Comment (abarker66) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/MemberService.java:51]: I think it should be removed; it served its purpose, the original refresh took approx 24 seconds and this version takes about 7 seconds.

**Review by LiamMacP — APPROVED** [review]
Approved - looks like a much better solution!

**Pull request merged**